From 2847293916b20869f8a88caa42c1af529229193b Mon Sep 17 00:00:00 2001
From: Oliver Kurth <okurth@gmail.com>
Date: Tue, 9 May 2023 15:59:13 -0700
Subject: [PATCH 04/14] shorten stack trace output for better visibility in
 case we hit an error

Example by adding and invalid option to the config file:

Unknown install_config keys: failme
Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/photon_installer/main.py", line 56, in main
    installer.configure(install_config)
  File "/usr/lib/python3.10/site-packages/photon_installer/installer.py", line 169, in configure
    raise Exception(issue)
Exception: Unknown install_config keys: failme

Change-Id: Ia73ee5855a20509d416fc4bd061bb668448e0d44
---
 photon_installer/main.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/photon_installer/main.py b/photon_installer/main.py
index c04c5ae..84c2873 100644
--- a/photon_installer/main.py
+++ b/photon_installer/main.py
@@ -4,6 +4,9 @@
 # */
 from os.path import dirname, join
 from argparse import ArgumentParser
+import sys
+import traceback
+
 
 def main():
     usage = "Usage: %prog [options]"
@@ -45,7 +48,9 @@ def main():
             installer.configure(install_config)
             installer.execute()
     except Exception as err:
-        raise Exception(f"Installer failed with error: {err}")
+        traceback.print_exc()
+        sys.exit(1)
+
 
 if __name__ == '__main__':
     main()
-- 
2.23.1

